homeassistant/components/androidtv/media_player.py

```diff
@@ -304,12 +310,24 @@ def adb_command(self, cmd):
         """Send an ADB command to an Android TV / Fire TV device."""
         key = self._keys.get(cmd)
         if key:
-            return self.aftv.adb_shell('input keyevent {}'.format(key))
+            self.aftv.adb_shell('input keyevent {}'.format(key))
+            self._adb_response = None
+            self.schedule_update_ha_state()
+            return
 
         if cmd == 'GET_PROPERTIES':
-            return self.aftv.get_properties_dict()
+            self._adb_response = str(self.aftv.get_properties_dict())
+            self.schedule_update_ha_state()
+            return self._adb_response
+
+        response = self.aftv.adb_shell(cmd)
+        if isinstance(response, str) and response.strip():
```

Comment:
This feels a bit hacky.. Is there some case where the adb_shell() call would be returning something else than a string? If not, it'd be just easier to check for that here? Otherwise I think this would be better handled in the backend lib.

Other than that it looks fine to me; I don't have a device at hand, so I cannot really do any testing on functionality.

Reply:
It's either a string or `None`. I've been using this version of the component, it works as expected.

Reply:
I think I'd personally prefer then checking against Noneness here, but it's fine as it is. Let's merge this, thanks! :tada: